KULeuvenNeuromechanics/PredSim
# Add implementation ligaments

### Updates
- Simulations now include contributions of ligaments to joint moments.
- Included 2 new example models
   * DHondt_2023_2seg: updated version of Falisse_et_al_2022 
   * DHondt_2023_3seg: more complex model with 3-segment foot model. This model has ligaments in the foot
- Created a folder ModelComponents
   * The idea is to make it easier to use custom components e.g. ligament force-length characteristics.
   * Included 3 ligament force-length functions and a template

### How to test
Run a simulation for both new models. Results should look like figure 2 in https://doi.org/10.1101/2023.03.22.533790
Recommended settings for 3-segment foot model are selected in main.m
For the 2-segment foot model, use
`S.subject.mtp_type = '2022paper';`
`S.subject.set_stiffness_coefficient_selected_dofs = {'mtp_angle',25};`
`S.subject.set_damping_coefficient_selected_dofs = {'mtp_angle',2};`
`S.subject.muscle_pass_stiff_shift = {{'soleus','_gas','per_','tib_','_dig_','_hal_'},0.9};`
